File: react/aaudiocore/src/models/audio-state.ts
export enum PlaybackStatus {
  Idle = "idle",
  Loading = "loading",
  Ready = "ready",
  Playing = "playing",
  Paused = "paused",
  Buffering = "buffering",
  Ended = "ended",
  Error = "error"
}

export interface AudioState {
  trackId: string | null;
  sourceUri: string | null;
  status: PlaybackStatus;
  currentMs: number;
  durationMs: number;
  volume: number;
  playbackRate: number;
  errorMessage?: string;
}

export const createInitialAudioState = (): AudioState => ({
  trackId: null,
  sourceUri: null,
  status: PlaybackStatus.Idle,
  currentMs: 0,
  durationMs: 0,
  volume: 1,
  playbackRate: 1
});

File: winforms/aaudiocore/Models/AudioState.cs
namespace AAudioCore.Models;

public sealed record AudioState
{
    public string? TrackId { get; init; }
    public string? SourceUri { get; init; }
    public PlaybackStatus Status { get; init; } = PlaybackStatus.Idle;
    public double CurrentMs { get; init; }
    public double DurationMs { get; init; }
    public float Volume { get; init; } = 1f;
    public float PlaybackRate { get; init; } = 1f;
    public string? ErrorMessage { get; init; }
}
